plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/DependencyExtractorService.kt

```diff
@@ -78,55 +76,63 @@ abstract class DependencyExtractorService :
         tailrec fun recursivelyExtractProjects(projects: Set<LoadProjectsBOT.Result.Project>) {
             if (projects.isEmpty()) return
             projects.forEach { project ->
-                gitHubRepositorySnapshotBuilder.addProject(details.buildPath, project.path, project.buildFile)
+                gitHubRepositorySnapshotBuilder.addProject(project.identityPath, project.buildFile)
             }
             val newProjects = projects.flatMap { it.children }.toSet()
             recursivelyExtractProjects(newProjects)
         }
         recursivelyExtractProjects(setOf(result.rootProject))
     }
 
-    private fun extractDependencies(
-        details: ResolveConfigurationDependenciesBOT.Details,
-        result: ResolveConfigurationDependenciesBOT.Result
+    private fun extractConfigurationDependencies(
+        details: ResolveConfigurationDependenciesBuildOperationType.Details,
+        result: ResolveConfigurationDependenciesBuildOperationType.Result
     ) {
-        val rootComponentId = result.rootComponent.id
-        val trueProjectPath =
-            (rootComponentId as? ProjectComponentIdentifier)?.projectPath
-                ?: details.projectPath
+        val repositoryLookup = RepositoryUrlLookup(details, result)
+        val rootComponent = result.rootComponent
+        val projectIdentityPath = (rootComponent.id as? DefaultProjectComponentIdentifier)?.identityPath?.path
         @Suppress("FoldInitializerAndIfToElvis") // Purely for documentation purposes
-        if (trueProjectPath == null) {
+        if (projectIdentityPath == null) {
             // TODO: We can actually handle this case, but it's more complicated than I have time to deal with
             // TODO: See the Gradle Enterprise Build Scan Plugin: `ConfigurationResolutionCapturer_5_0`
             return
         }
 
-        val repositoryLookup = RepositoryUrlLookup(details, result)
-        val dependencies =
-            extractDependenciesFromResolvedComponentResult(
-                result.rootComponent,
-                repositoryLookup::doLookup
-            )
-        // Feel free to change this. This is free-form and has no spec associated with it.
-        val metaData = mapOf(
-            "project_path" to details.projectPath,
-            "configuration" to details.configurationName,
-            "build_path" to details.buildPath,
-            "configuration_description" to details.configurationDescription,
-            "is_script_configuration" to details.isScriptConfiguration,
-            "true_project_path" to trueProjectPath
-        )
```

Comment:
This meta-data was included to help provide further context around what project produced which output.

If it's possible to still capture this information and encode it into the GitHub model that would be nice, especially for external tools reading from GitHub's API and trying to map that data to actual files.

Reply:
At this stage, I want to keep things as simple as possible until we have a first working version. It will be possible to add further metadata at a later stage if it proves valuable.

> external tools reading from GitHub's API and trying to map that data to actual files
I wasn't aware of the ability to read the original data from GitHub. Can you point to the docs?

Reply:
Everything labled `metadata`
https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28